llvm/lib/Transforms/InstCombine/InstCombineAddSub.cpp

```diff
@@ -2631,6 +2631,21 @@ Instruction *InstCombinerImpl::visitSub(BinaryOperator &I) {
                                                /* IsNUW */ false))
       return replaceInstUsesWith(I, Res);
 
+  if (match(Op0, m_ZExt(m_PtrToInt(m_Value(LHSOp)))) &&
```

Comment:
```suggestion
  if (match(Op0, m_ZExt(m_PtrToIntSameSize(DL, m_Value(LHSOp)))) &&
```
Counterexample: https://alive2.llvm.org/ce/z/BJujTw
Please add a negative test as well.

Reply:
That's quite interesting case. Will add your example to negatives. Thanks